solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkUtility.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.common;
+
+import co.cask.cdap.api.data.schema.Schema;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+
+/**
+ * Utility class for SolrSearchSink and RealtimeSolrSearchSink plugins.
+ */
+public class SolrSearchSinkUtility {
+
+  private static final String SINGLE_NODE_SOLR_MODE = "Single Node Solr";
+  private static final String SOLR_CLOUD_MODE = "SolrCloud";
+
+  /**
+   * Returns the SolrClient for establishing the connection with Solr server, using the hostname and port provided by
+   * user.
+   *
+   * @return Solr client
+   */
+  public static SolrClient getSolrConnection(SolrSearchSinkConfig config) {
+    String urlString;
+    SolrClient solrClient = null;
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE)) {
+      urlString = "http://" + config.getSolrHost() + "/solr/" + config.getCollectionName();
+      solrClient = new HttpSolrClient(urlString);
+    } else if (config.getSolrMode().equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(config.getSolrHost());
+      solrCloudClient.setDefaultCollection(config.getCollectionName());
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  public static void validateSolrConnectionString(SolrSearchSinkConfig config) {
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE) && config.getSolrHost().contains(",")) {
```

Comment:
There is a possibility that user can run SolrCloud on one zookeeper (one host) only, may be for testing purposes. In this case, then it will be difficult to identify whether the single host is for SolrCloud or Single Node. That’s why I kept the mode setting.

Assuming that if we will always get multiple hosts for SorCloud, then we can remove the mode setting.  Let me know your thoughts on this.